chromium/browser_type.go

```diff
@@ -142,14 +142,19 @@ func (b *BrowserType) Launch(opts goja.Value) api.Browser {
 
 	flags := prepareFlags(launchOpts, &state.Options)
 
-	finalDirPath, cleanupFunc, err := b.createTempDir(flags["user-data-dir"], logger)
-	if err != nil {
-		k6common.Throw(rt, err)
+	dataDir := &b.storage
+	if err := dataDir.Make("", flags["user-data-dir"]); err != nil {
+		k6common.Throw(rt, fmt.Errorf("cannot make temp data directory: %w", err))
 	}
-	flags["user-data-dir"] = finalDirPath
+	flags["user-data-dir"] = dataDir.Dir
 
 	go func(ctx context.Context) {
-		defer cleanupFunc()
+		defer func() {
+			err := dataDir.Cleanup()
+			if err != nil {
+				logger.Errorf("BrowserType:Launch", "%v", err)
+			}
+		}()
```

Comment:
```suggestion
			if err := dataDir.Cleanup(); err != nil {
				logger.Errorf("BrowserType:Launch", "%s", err)
			}
```

Reply:
Resolved in commit: 15022675046f46849db9e5f3c3ae3f54b1683cfa